pkg/agent/multicluster/stretched_networkpolicy_controller.go

```diff
@@ -0,0 +1,347 @@
+	}
+	s.lock.Unlock()
+
+	labelID := openflow.UnknownLabelIdentity
+	if objs, err := s.labelIdentityInformer.Informer().GetIndexer().ByIndex(labelIndex, normalizedLabel); err == nil && len(objs) == 1 {
+		labelIdentity := objs[0].(*v1alpha1.LabelIdentity)
+		labelID = labelIdentity.Spec.ID
+	}
+	return labelID
+}
+
+func (s *StretchedNetworkPolicyController) processPodCNIAddEvent(e interface{}) {
+	podEvent := e.(antreatypes.PodUpdate)
+	if !podEvent.IsAdd {
+		return
+	}
+	podRef := types.NamespacedName{
+		Namespace: podEvent.PodNamespace,
+		Name:      podEvent.PodName,
+	}
+	s.queue.Add(podRef)
+}
+
+// processPodUpdate handles Pod update events. It only enqueues the Pod if the
+// Labels of this Pod have been updated.
+func (s *StretchedNetworkPolicyController) processPodUpdate(old, cur interface{}) {
+	oldPod, _ := old.(*v1.Pod)
+	curPod, _ := cur.(*v1.Pod)
+	if curPod.Spec.HostNetwork {
+		klog.V(5).InfoS("skipped processing hostNetwork Pod update event", "name", curPod.Name, "namespace", curPod.Namespace)
+		return
+	}
+	if reflect.DeepEqual(oldPod.Labels, curPod.Labels) {
+		klog.V(5).InfoS("Pod UpdateFunc received a Pod update event, "+
+			"but labels are the same. Skip it", "name", curPod.Name, "namespace", curPod.Namespace)
+		return
+	}
+	s.queue.Add(getPodReference(curPod))
+}
+
+// processPodDelete handles Pod delete events. It deletes the Pod from the
+// labelToPods and podToLabel. After Pod is deleted, its classifier flow will also
+// be deleted by podConfigurator. So no need to enqueue this Pod to update its
+// classifier flow.
+func (s *StretchedNetworkPolicyController) processPodDelete(old interface{}) {
+	oldPod, _ := old.(*v1.Pod)
+	oldPodRef := getPodReference(oldPod)
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	podLabel := s.podToLabel[oldPodRef]
+	s.deleteLabelToPod(podLabel, oldPodRef)
+	delete(s.podToLabel, oldPodRef)
+}
+
+// processNamespaceUpdate handles Namespace update events. It only enqueues all
+// Pods in this Namespace if the Labels of this Namespace have been updated.
+func (s *StretchedNetworkPolicyController) processNamespaceUpdate(old, cur interface{}) {
+	oldNS, _ := old.(*v1.Namespace)
+	curNS, _ := cur.(*v1.Namespace)
+	if reflect.DeepEqual(oldNS.Labels, curNS.Labels) {
+		klog.V(5).InfoS("Namespace UpdateFunc received a Namespace update event, but labels are the same. Skip it", "namespace", curNS.Name)
+		return
+	}
+	allPodsInNS, _ := s.podLister.Pods(curNS.Name).List(labels.Everything())
+	for _, pod := range allPodsInNS {
+		if pod.Spec.HostNetwork {
+			continue
+		}
+		s.queue.Add(getPodReference(pod))
+	}
+}
+
+// processLabelIdentityEvent handles labelIdentity add/update/delete event.
+// It will enqueue all Pods affected by this labelIdentity
+func (s *StretchedNetworkPolicyController) processLabelIdentityEvent(cur interface{}) {
+	labelIdentity, _ := cur.(*v1alpha1.LabelIdentity)
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	if podSet, ok := s.labelToPods[labelIdentity.Spec.Label]; ok {
```

Comment:
just to confirm, a labelIdentity's label never changes, right?

Reply:
Yes. Different label will be treated as a different labelIdentity.